plugin/query_predicates.lua

```diff
@@ -6,14 +6,25 @@ local query = vim.treesitter.query
 ---@param pred string[]
 ---@return boolean|nil
 query.add_predicate('has-type?', function(match, _, _, pred)
-  local node = match[pred[2]]
-  if not node then
+  local nodes = match[pred[2]]
+  if type(nodes) ~= 'table' then
+    nodes = { nodes }
+  end
+  if not nodes or #nodes == 0 then
     return true
   end
 
   local types = { unpack(pred, 3) }
-  return vim.list_contains(types, node:type())
-end, true)
+  for _, node in ipairs(nodes) do
+    if vim.list_contains(types, node:type()) then
+      return true
+    end
+  end
+  return false
```

Comment:
Isn't this the behavior of `any-has-type?` i.e. only need 1 node of the quantifiers to match.

Reply:
Yes you're right, good catch. I'll add a separate `any-has-type?` predicate for the "any" behavior.